easybuild/tools/filetools.py

```diff
@@ -460,26 +465,47 @@ def download_file(filename, url, path, forced=False):
     attempt_cnt = 0
 
     # use custom HTTP header
-    url_req = urllib2.Request(url, headers={'User-Agent': 'EasyBuild',  "Accept" : "*/*"})
+    headers = {'User-Agent': 'EasyBuild', 'Accept': '*/*'}
+    # for backward compatibility, and to avoid relying on 3rd party Python library 'requests'
+    url_req = urllib2.Request(url, headers=headers)
+    used_urllib = urllib2
```

Comment:
@bartoldeman I'd prefer using a boolean here (`use_urllib2`) and handling `HTTPError` via the `import` above

Reply:
I cannot `from requests.exceptions import HTTPError` before it is actually used (because at the first download attempt `HTTPError` is still `urllib.HTTPError`, that is why
I changed to `used_urllib`. If I do both:
```
use_urllib2 = True
HTTPError = requests.HTTPError
```
that's a bit double...

Perhaps I should just write a `urlopen()` emulation using requests (including raising the urllib2.HTTPError exception manually), which might be the cleanest then.

Reply:
Right, I overlooked that you first need `HTTPError` from `urllib2`...